Fix behavior of getRemainingTimeInMillis in local invokation

Nevon opened the pull request:
<!--
1. Please check out and follow our Contributing Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
2. Do not remove any section of the template. If something is not applicable leave it empty but leave it in the PR
3. Please follow the template, otherwise we'll have to ask you to update it and it will take longer until your PR is merged
-->

## What did you implement:

The behavior of `context.getRemainingTimeInMillis` is incorrect when invoking a function locally. It is supposed to return the amount of milliseconds remaining until the timeout of the function is reached, but in actuality it's just returning the number of milliseconds since the function was invoked.

I discovered this when my workers were behaving really strangely when invoked locally. They would just pick a single item from the queue and then stop, because the time remaining would always be <1000, which I had set as the limit for when to stop working.

I have now fixed that behavior so that it counts down from the timeout instead of up from the start time.

## How did you implement it:

Simply return the timeout in milliseconds minus the elapsed time since starting.

## How can we verify it:

Run the tests. I couldn't find a good way to fake the timer, since I didn't find a way to replace the date constructor, so I just added a 1ms blocking loop in those tests.

You could also just invoke a function locally and log `context.getTimeRemainingInMillis()`. Before, you would have seem something like `5`, and now you should see something around `5995` (unless you have specified a different timeout).

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

RafalWilinski commented:
Thanks @Nevon for your first PR 🎉 

Python part is not tested at all, see `invokeLocalPython` function's coverage info. Besides that, everything is looking good from my perspective.

Nevon commented:
Yeah, nothing about that Python implementation appears to have been tested previously. I added at least some rudimentary testing of the part that I've implemented, to at least please the coverage gods. It looks to me that the Python implementation is a work in progress, but I've never used Serverless in Python, so I'm not sure what the expected level of support is.

Nevon commented:
Ah crap, now I know why that's not tested at all. When the tests are running in CI, of course the container doesn't have python, so that test is not going to pass there.

I'm going to mark it as skipped for now, because AFAIK, travis doesn't currently support multi languages in a good way (you can hack around it, but it will slow the build down tremendously).

eahefnawy commented:
@Nevon this is beautiful! ❤️ ... Thanks a lot for that! We're always trying to emulate Lambda locally as accurately as possible.

G2M! ✨
